setup.cfg

```diff
@@ -20,7 +20,7 @@ classifiers =
 packages = find:
 install_requires =
     PyCifRW~=4.4
-    aiida-core>=2.1,<3
+    aiida-core>=2.5,<3
```

Comment:
We should be very careful about bumping the minimum aiida-core version since it might cause issues for downstream apps. If at all possible, please make the ipyoptimade compatible with earlier aiida-core versions. Ideally, we should allow pip to pick the correct version based on which aiida-core is already installed in the image.

Reply:
Good point, I think very hard and here is some thought.

`ipyoptimade` has nothing to do with `aiida-core` but with `pydantic`.
- `ipyoptimade<0.2.0` which right after I made the migration, it only support `pydantic==v1.0`
- `ipyoptimade==0.2.0` is the version that support pydantic v2.

So if we need something that need `aiida-core==2.5.0` then need `pydantic` v2 is the environment and thus need to use `ipyoptimade==0.2.0`. 
The problem now comes with that the use of optimade widget comes from `aiidalab-widgets-base` and it is another base dependencies in an AiiDAlab environment.

One workaround is not use optimade widget from `aiidalab-widgets-base` but from `ipyoptimade` directly.Then nothing need to change here. 
Maybe another workaround worth to try is what you mentioned, that the `ipyoptimade` is an extra dependency. 

In summary, I think we can not guarantee that one app not break another app. But we need to keep the bottom line that we need to try our best the new version of aiidalab docker-stack will not break the app.

Reply:
It would be very helpful if you have a clear "break path" like which update of which app or which component would cause the break of which part of an app.

Reply:
I think our general goal with AWB is to be as independent of various other core dependencies as possible, to avoid coupling between versions of different dependencies. 

>   ipyoptimade<0.2.0 which right after I made the migration, it only support pydantic==v1.0
    ipyoptimade==0.2.0 is the version that support pydantic v2.

This is great that there are diifferent versions that support different pydantic versions! In the best case scenario, we should allow both versions in AWB and let `pip` to pick the correct version. Whether that will work in practice is unclear to me, since pip can ignore the existing aiida-core version constraints. This is another reason why we really should publish AWB as conda package and install it together with `aiida-core` in the base image.